astrokat/observe_main.py

```diff
@@ -220,14 +220,34 @@ def cadence_target(target_list):
 
 def above_horizon(katpt_target, horizon=20.0, duration=0.0):
     """Check target visibility."""
-    [azim, elev] = katpt_target.azel(time.time())
+    # use local copies so you do not overwrite target time attribute
+    if 'azel' in katpt_target.tags:
+        return True
```

Comment:
Sorry, incomplete test -- updated to include desired functionality
```
if katpt_target.body.alt < horizon:
    return False
else:
    return True
```

Since altitude is given all that is needed is a direct check
No date computation is needed, function exits quickly